front/components/agent_builder/capabilities/mcp/formValidation.ts

```diff
@@ -0,0 +1,200 @@
+      (val) => val !== null,
+      {
+        message: "Please select a Dust app",
+      }
+    );
+  } else {
+    dynamicFields.dustAppConfiguration = z.null();
+  }
+
+  if (
+    requirements.requiredStrings.length > 0 ||
+    requirements.requiredNumbers.length > 0 ||
+    requirements.requiredBooleans.length > 0 ||
+    Object.keys(requirements.requiredEnums).length > 0
+  ) {
+    // We just check if the key exists and don't check the value type.
+    dynamicFields.additionalConfiguration = additionalConfigurationSchema
+      .default({})
+      .refine(
+        (additionalConfig) => {
+          for (const key of requirements.requiredStrings) {
+            if (
+              !additionalConfig[key] 
+            ) {
+              return false;
+            }
+          }
+
+          for (const key of requirements.requiredNumbers) {
+            if (
+              additionalConfig[key] === undefined
+            ) {
+              return false;
+            }
+          }
+
+          for (const key of requirements.requiredBooleans) {
+            if (
+              additionalConfig[key] === undefined
+            ) {
+              return false;
+            }
+          }
+
+          for (const [key] of Object.entries(
+            requirements.requiredEnums
+          )) {
+            if (
+              !additionalConfig[key]
+            ) {
+              return false;
+            }
+          }
+
+          return true;
+        },
+        {
+          message: "All required configuration fields must be filled",
+        }
+      );
+  } else {
+    dynamicFields.additionalConfiguration =
+      additionalConfigurationSchema.default({});
+  }
+
+  const configSchema = z.object({
+    ...baseFields,
+    ...dynamicFields,
+  });
+
+  return configSchema;
+};
+
+// All MCP actions have the same config structure, but required fields are different and we need to determine them from mcpServerView.
+export function getMCPConfigurationFormSchema(
+  mcpServerView: MCPServerViewType | null | undefined
+) {
+  const requirements = mcpServerView
+    ? getMCPServerRequirements(mcpServerView)
+    : null;
+
+  return z.object({
+    name: z
+      .string()
+      .min(1, "The name cannot be empty.")
+      .regex(
+        /^[a-z0-9_]+$/,
+        "The name can only contain lowercase letters, numbers, and underscores (no spaces)."
+      )
+      .default(""),
+    description: z
+      .string()
+      .min(1, "Description is required")
+      .max(800, "Description too long"),
+    configuration: getConfigurationSchema(requirements),
+  });
+}
+
+export function getDefaultConfiguration(
+  mcpServerView: MCPServerViewType | null | undefined
+) {
+  const requirements = mcpServerView
+    ? getMCPServerRequirements(mcpServerView)
+    : null;
+
+  const defaults: MCPServerConfigurationType = {
+    mcpServerViewId: mcpServerView?.sId ?? "not-a-valid-sId",
```

Comment:
🤔

Reply:
I have no idea what it is, it's taken from assistant builder 🤔